Lua/Lua.sublime-syntax

```diff
@@ -8,18 +8,16 @@ file_extensions:
 scope: source.lua
 contexts:
   main:
-    - match: '\b(function)\b\s*([a-zA-Z_]\w*)?\s*(([.]\s*[a-zA-Z_]\w*\s*)*)\s*(([.:])?\s*[a-zA-Z_]\w*)?\s*(\()([^)]*)(\))?'
+    - match: '\b(function)\b\s*(([a-zA-Z_]\w*\s*[.]\s*)*)?\s*([a-zA-Z_]\w*\s*[:])?\s*([a-zA-Z_]\w*)?\s*(\()([^)]*)(\))?'
       scope: meta.function.lua
       captures:
         1: keyword.control.lua
         2: entity.name.function.scope.lua
-        3: entity.name.function.scope.lua
```

Comment:
If you don't want to assign a scope to a group, you should make it non-capturing with `(?:)`.

Reply:
But i want it to capture:
```
function foo.bar:fooo(...)

end
```
`bar:` here is only a scope, and not a name.